front/src/common/osrdContext.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { createContext, useContext, useMemo } from 'react';
+
+import type { ObjectFieldsTypes } from 'utils/types';
+
+import { MODES } from 'applications/operationalStudies/consts';
+
+import { ModalProvider } from 'common/BootstrapSNCF/ModalSNCF/ModalProvider';
+
+import type { OsrdSlicesType } from 'reducers';
+import type { editorSelectorsType } from 'reducers/editor/selectors';
+import type { mapViewerSelectorsType } from 'reducers/mapViewer/selectors';
+import type { stdcmConfSelectorsType } from 'reducers/osrdconf/stdcmConf/selectors';
+import type { operationalStudiesConfSelectorsType } from 'reducers/osrdconf/operationalStudiesConf/selectors';
+
+export type OsrdSelectorsType =
+  | operationalStudiesConfSelectorsType
+  | stdcmConfSelectorsType
+  | mapViewerSelectorsType
+  | editorSelectorsType;
+
+export type OsrdContextType = {
+  slice: OsrdSlicesType;
+  selectors: OsrdSelectorsType;
+  mode: ObjectFieldsTypes<typeof MODES>;
+  isStdcm: boolean;
+  isSimulation: boolean;
+} | null;
+
+const OsrdContext = createContext<OsrdContextType>(null);
+
+export const useOsrdContext = () => {
+  const context = useContext(OsrdContext);
+  if (!context) {
+    throw new Error('useOsrdContext must be used within a OsrdContext.Provider');
+  }
+  return context;
+};
+
+export const useOsrdActions = () => {
+  const { slice } = useOsrdContext();
+  if (!slice) {
+    throw new Error('OsrdContext slice is not available');
+  }
+
+  if (!slice.actions) {
+    throw new Error('OsrdContext slice does not have actions');
+  }
+
+  return slice.actions;
+};
+
+export const useOsrdSelectors = () => {
+  const { selectors } = useOsrdContext();
+  if (!selectors) {
+    throw new Error('OsrdContext selectors are not available');
+  }
+
+  return selectors;
+};
+
+type OsrdContextLayoutProps = {
+  slice: NonNullable<OsrdContextType>['slice'];
+  selectors: NonNullable<OsrdContextType>['selectors'];
+  mode: ObjectFieldsTypes<typeof MODES>;
+};
+
+export const OsrdContextLayout = ({ slice, selectors, mode }: OsrdContextLayoutProps) => {
+  const isStdcm = mode === MODES.stdcm;
+  const isSimulation = mode === MODES.simulation;
+  const value = useMemo(
+    () => ({ slice, selectors, mode, isStdcm, isSimulation }),
+    [slice, selectors]
+  );
+  return (
+    <OsrdContext.Provider value={value}>
+      <ModalProvider />
```

Comment:
why is this modal the only one to receive the context'

Reply:
@kmer2016

Reply:
I didn't remember why we have do like this @Math-R

Reply:
twas a problem with ModalContext that can’t be above the router in the tree, all good